apps/explorer/lib/explorer/chain.ex

```diff
@@ -5141,6 +5141,91 @@ defmodule Explorer.Chain do
     end
   end
 
+# tokenx_by_address_and_id is based on the following SQL query:
+#
+#   select
+#     tokens.contract_address_hash as token_address,
+#     ti.token_id,
+#     (select tt.block_number
+#      from token_transfers tt
+#      where ti.token_id = tt.token_id and tokens.contract_address_hash = tt.token_contract_address_hash
+#        and from_address_hash = E'\\x0000000000000000000000000000000000000000' limit 1) as block_number_minted,
+#     CASE WHEN ti.token_id IS NULL
+#      THEN total_supply
+#      ELSE 1
+#     END AS amount,
```

Comment:
This part looks a little suspicious to me. Since we are calling this function passing a concrete token_id it is very unlikely it is going to be null here, and therefore the amount will always be 1. Is that expected? (This PR at the moment sets amount to be equal to total_supply from the tokens table).